### ISSUE TITLE
Sidecar proxy for fortio fails readiness check

### ISSUE THREAD

**Issue by vishal-yadav** [istio/istio#11888]
**Describe the bug**
Deployment of fortio as per [Adding a client](https://preliminary.istio.io/docs/tasks/traffic-management/circuit-breaking/#adding-a-client) section for circuit breaker verification task, fails with readiness probe check.
```
$ kubectl describe pod fortio-deploy-98d99b6d-kgkgp
...
    State:          Running
      Started:      Wed, 20 Feb 2019 16:43:10 +0530
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:      10m
    Readiness:  http-get http://:15020/healthz/ready delay=1s timeout=1s period=2s #success=1 #failure=30
    Environment:
      POD_NAME:                      fortio-deploy-98d99b6d-kgkgp (v1:metadata.name)
      POD_NAMESPACE:                 default (v1:metadata.namespace)
      INSTANCE_IP:                    (v1:status.podIP)
      ISTIO_META_POD_NAME:           fortio-deploy-98d99b6d-kgkgp (v1:metadata.name)
      ISTIO_META_CONFIG_NAMESPACE:   default (v1:metadata.namespace)
      ISTIO_META_INTERCEPTION_MODE:  REDIRECT
      ISTIO_METAJSON_LABELS:         {"app":"fortio"}
...
  Normal   Pulled     50m                   kubelet, workerblr  Container image "docker.io/istio/proxyv2:1.1.0-snapshot.6" already present on machine
  Normal   Created    50m                   kubelet, workerblr  Created container
  Normal   Started    50m                   kubelet, workerblr  Started container
  Normal   Scheduled  50m                   default-scheduler   Successfully assigned default/fortio-deploy-98d99b6d-kgkgp to workerblr
  Warning  Unhealthy  26s (x1498 over 50m)  kubelet, workerblr  Readiness probe failed: HTTP probe failed with statuscode: 503
```
```
$ kubectl logs fortio-deploy-98d99b6d-kgkgp -c istio-proxy
...
2019-02-20T11:13:10.036610Z     info    Envoy command: [-c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 45 --parent-shutdown-time-s 60 --service-cluster fortio.default --service-node sidecar~10.40.0.40~fortio-deploy-98d99b6d-kgkgp.default~default.svc.cluster.local --max-obj-name-len 189 --allow-unknown-fields -l warning --concurrency 1]
...
2019-02-20T12:05:04.364498Z     info    Envoy proxy is NOT ready: 5 errors occurred:

* failed checking application ports. listeners="0.0.0.0:15090","10.98.156.79:443","10.96.114.76:15011","10.98.156.79:31400","10.98.156.79:15029","10.98.156.79:15031","10.96.0.1:443","10.97.130.92:443","10.97.130.92:15443","10.98.156.79:15443","10.108.21.109:443","10.98.156.79:15030","10.98.156.79:15032","0.0.0.0:8080","0.0.0.0:15004","0.0.0.0:80","0.0.0.0:15014","0.0.0.0:15010","0.0.0.0:9091","0.0.0.0:8000","10.40.0.40:15020","0.0.0.0:15001"
* envoy missing listener for inbound application port: 0
* envoy missing listener for inbound application port: 0
* envoy missing listener for inbound application port: 8080
* envoy missing listener for inbound application port: 8079
```
```
$ istioctl proxy-status
Warning! error execing into istio-pilot-5b49dc5f77-65p4z/istio-system discovery container: gc 1 @0.012s 10%: 0.042+1.2+1.5 ms clock, 0.17+0.14/0/2.2+6.2 ms cpu, 4->4->2 MB, 5 MB goal, 4 P
gc 2 @0.020s 14%: 0.006+1.9+1.3 ms clock, 0.026+0.14/1.8/0.059+5.4 ms cpu, 4->4->3 MB, 5 MB goal, 4 P

NAME                                                   CDS        LDS        EDS               RDS          PILOT                            VERSION
fortio-deploy-98d99b6d-kgkgp.default                   SYNCED     SYNCED     SYNCED (50%)      SYNCED       istio-pilot-5b49dc5f77-65p4z     1.1.0
httpbin-744c9f787-kthhm.default                        SYNCED     SYNCED     SYNCED (50%)      SYNCED       istio-pilot-5b49dc5f77-65p4z     1.1.0
istio-egressgateway-d9f69d6c8-qnjqj.istio-system       SYNCED     SYNCED     SYNCED (100%)     NOT SENT     istio-pilot-5b49dc5f77-65p4z     1.1.0
istio-ingress-7795875fdd-ndtzp.istio-system            SYNCED     SYNCED     SYNCED (50%)      NOT SENT     istio-pilot-5b49dc5f77-65p4z     1.1.0
istio-ingressgateway-598c7757bc-9kh74.istio-system     SYNCED     SYNCED     SYNCED (100%)     NOT SENT     istio-pilot-5b49dc5f77-65p4z     1.1.0
```

**Expected behavior**
Fortio as client deployment should succeed. 

**Steps to reproduce the bug**
Follow [these](https://preliminary.istio.io/docs/tasks/traffic-management/circuit-breaking/#adding-a-client) steps to deploy fortio client.

**Version**
$ istioctl version
version.BuildInfo{Version:"1.1.0-snapshot.6", GitRevision:"14777199b85ba56ebca2a4516afff33c97199eb4", User:"root", Host:"f0f1db73-2fb5-11e9-86e9-0a580a2c0304", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.1.0-snapshot.5-156-g1477719"}

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.3", GitCommit:"435f92c719f279a3a67808c80521ea17d5715c66", GitTreeState:"clean", BuildDate:"2018-11-26T12:57:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.3", GitCommit:"435f92c719f279a3a67808c80521ea17d5715c66", GitTreeState:"clean", BuildDate:"2018-11-26T12:46:57Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}

**Installation**
Following [Minimal Install](https://preliminary.istio.io/docs/setup/kubernetes/minimal-install/) along with 
istio-egressgateway, istio-ingress, istio-ingressgateway and istio-policy components.

```
$ kubectl get pod,svc -n istio-system
NAME                                        READY   STATUS    RESTARTS   AGE
pod/istio-egressgateway-d9f69d6c8-94t7b     1/1     Running   0          2d1h
pod/istio-ingress-7795875fdd-l7ftc          1/1     Running   0          2d
pod/istio-ingressgateway-598c7757bc-cql2d   1/1     Running   0          2d1h
pod/istio-pilot-5b49dc5f77-7kmpq            1/1     Running   0          2d1h
pod/istio-policy-7d55f4994b-z7wh5           2/2     Running   0          2d1h

NAME                           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                                                                                                      AGE
service/istio-egressgateway    NodePort    10.98.17.101     <none>        80:32740/TCP,443:30704/TCP,15443:31079/TCP                                                                                   2d1h
service/istio-ingress          NodePort    10.106.171.154   <none>        80:32000/TCP,443:32221/TCP                                                                                                   2d
service/istio-ingressgateway   NodePort    10.101.243.210   <none>        80:31380/TCP,443:31390/TCP,31400:31400/TCP,15029:30021/TCP,15030:31091/TCP,15031:31673/TCP,15032:30405/TCP,15443:31667/TCP   2d1h
service/istio-pilot            ClusterIP   10.109.52.67     <none>        15010/TCP,15011/TCP,8080/TCP,15014/TCP                                                                                       2d1h
service/istio-policy           ClusterIP   10.107.207.108   <none>        9091/TCP,15004/TCP,15014/TCP                                                                                                 2d1h
```

**Environment**
Local Kubernetes Cluster on Ubuntu (not minikube)

**Cluster state**
NA

**Comment by vishal-yadav** [comment]
Detailed logs [here](https://pastebin.com/KMcgZFPe) for fortio-deploy pod.

**Comment by costinm** [comment]
@nmittler  seems to be readiness related.

Having it fail readiness once can happen - but normally it should recover as config becomes consistent.

**Comment by nmittler** [comment]
It looks like it's the zero port issue, which was just fixed by #12001

**Comment by duderino** [comment]
Closing as a duplicate of https://github.com/istio/istio/pull/12001
